app/code/community/Riskified/Full/Model/Observer/Order/Payment/Failed.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/**
+ * Riskified Full failed order payment observer.
+ *
+ * @category Riskified
+ * @package  Riskified_Full
+ * @author   Piotr Pierzak <piotrek.pierzak@gmail.com>
+ */
+class Riskified_Full_Model_Observer_Order_Payment_Failed
+{
+    /**
+     * Default failed transaction handler.
+     *
+     * @param Varien_Event_Observer $observer Observer object.
+     *
+     * @return Riskified_Full_Model_Observer_Order_Payment_Failed
+     */
+    public function handleDefaultFailedTransaction(
+        Varien_Event_Observer $observer
+    ) {
+        $order = $observer->getEvent()->getOrder();
+
+        $orderPaymentHelper = Mage::helper('full/order_payment');
+        $paymentDetails = $orderPaymentHelper->getPaymentDetails($order);
+
+        $paymentDetailsArray = json_decode($paymentDetails->toJson(), true);
```

Comment:
can you please extract this code (that gets the model data) to the helper as done in the other cases?